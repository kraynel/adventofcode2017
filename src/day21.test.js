const { initialGrid, getNextGrid, parseInput } = require('./day21');
const fs = require('fs');

describe('getNextGrid', () => {
  //   it('should getNextGrid', () => {
  //     const transformations = parseInput(`../.# => ##./#../...
  // .#./..#/### => #..#/..../..../#..#`);
  //     expect(getNextGrid(initialGrid, transformations)).toEqual([
  //       '#..#',
  //       '....',
  //       '....',
  //       '#..#'
  //     ]);
  //   });

  //   it('should getNextGrid', () => {
  //     const transformations = parseInput(`../.# => ##./#../...
  // .#./..#/### => #..#/..../..../#..#`);
  //     expect(
  //       getNextGrid(['....', '.#.#', '....', '.#.#'], transformations)
  //     ).toEqual(['##.##.', '#..#..', '......', '##.##.', '#..#..', '......']);
  //   });

  it('should get the value', done => {
    fs.readFile('inputs/day21.txt', 'utf8', (err, data) => {
      const transformations = parseInput(data);

      const grids = [initialGrid];
      while (grids.length < 6) {
        console.log(`GRID ${grids.length - 1}`, grids[grids.length - 1]);
        grids.push(getNextGrid(grids[grids.length - 1], transformations));
      }

      console.log(`GRID ${grids.length - 1}`, grids[grids.length - 1]);

      const finalGrid = grids[grids.length - 1];
      expect(
        finalGrid
          .join('')
          .split('')
          .filter(a => a === '#').length
      ).toMatchSnapshot();
      done();
    });
  });
});
