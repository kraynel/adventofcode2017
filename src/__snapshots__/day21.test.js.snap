// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getNextGrid should get the value 1`] = `171`;
